stdlib/public/Concurrency/MainActor.swift

```diff
@@ -99,3 +99,41 @@ extension MainActor {
   }
 }
 #endif
+
+/// Performs a runtime test to check whether this function was called
+/// while on the MainActor. Then the operation is invoked and its
+/// result is returned.
+///
+/// - Attention:
+/// This operation is unsafe because if the runtime check fails, the
+/// operation may still be invoked off of the MainActor! You can control
+/// the behavior of check failure by setting the environment variable
+/// `SWIFT_UNEXPECTED_EXECUTOR_LOG_LEVEL` as follows:
+///
+///   - 0 ignores check failures
+///   - 1 will only log a warning (default)
+///   - 2 means fatal error
+///
+/// When in modes other than `0`, a message is output to standard error.
+///
```

Comment:
I think we'll need to polish this up a bit in evolution, for example the "log a warning" mode cannot be as reliably implemented as the precondition one, because we may be on a non-main thread but on the main queue (`main.sync { <check here> }` that may not be on the main thread, but is on the main queue). 

But I think the general shape of this is good and it is worth iterating on the implementation a bit more :)

I'm thinking if we might end up with a safe assume that asserts and an unsafe assume that is going to be mode 0 or 1, and the 1 mode can yield false-negatives in some rare cases we would need to document.

Overall though, definitely a worthy addition 👍